addons/stock/models/stock_picking.py

```diff
@@ -862,6 +862,15 @@ def do_transfer(self):
         """ If no pack operation, we do simple action_done of the picking.
         Otherwise, do the pack operations. """
         # TDE CLEAN ME: reclean me, please
+
+        # Double check that the pick is in the state that allows processing
+        # to avoid the creation of unintended Extra Moves
+        for pick in self:
+            if pick.state == 'done':
+                raise UserError(_('The pick is already validated'))
+            if pick.state == 'cancel':
+                raise UserError(_('The pick has been cancelled'))
```

Comment:
Can it be written in a bit simpler way?  if self.filtered(lambda p: p.state in ('done', 'cancel')):